src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -80,7 +86,34 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             // No AT in the cache or AT needs to be refreshed
             try
             {
-                return await FetchNewAccessTokenAsync(cancellationToken).ConfigureAwait(false);
+                if (cachedAccessTokenItem == null)
+                {
+                    return await FetchNewAccessTokenAsync(cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    if (cachedAccessTokenItem.NeedsRefresh())
+                    {
+                        _ = Task.Run(async () =>
+                        {
+                            // TODO - Exception handling is being discussed.
+                            try
+                            {
+                                await FetchNewAccessTokenAsync(cancellationToken).ConfigureAwait(false);
+                            }
+                            catch (MsalServiceException ex)
+                            {
+                                logger.Warning($"RefreshRtOrFailAsync Refreshing the RT failed. Is AAD down? { ex.IsAadUnavailable()}. Error {ex.Message} ");
```

Comment:
Generally try to log the entire message. ex.ToString() prints message + stack trace.